python-checks/src/main/java/org/sonar/python/checks/FloatingPointEqualityCheck.java

```diff
@@ -38,20 +43,29 @@
 public class FloatingPointEqualityCheck extends PythonSubscriptionCheck {
 
   private static final String MESSAGE = "Do not perform equality checks with floating point values.";
-  private static final String QUICK_FIX_MESSAGE = "Replace with %smath.isclose().";
+  private static final String QUICK_FIX_MESSAGE = "Replace with %s%s.isclose().";
 
-  private static final String QUICK_FIX_REPLACE = "%smath.isclose(%s, %s, rel_tol=1e-09, abs_tol=1e-09)";
+  private static final String QUICK_FIX_MATH = "%s%s.isclose(%s, %s, rel_tol=1e-09, abs_tol=1e-09)";
+
+  private static final String QUICK_FIX_IMPORTED_MODULE = "%s%s.isclose(%s, %s, rtol=1e-09, atol=1e-09)";
 
   private static final Tree.Kind[] BINARY_OPERATION_KINDS = { Tree.Kind.PLUS, Tree.Kind.MINUS, Tree.Kind.MULTIPLICATION,
       Tree.Kind.DIVISION };
 
   private ReachingDefinitionsAnalysis reachingDefinitionsAnalysis;
+  private static final List<String> SUPPORTED_IS_CLOSE_MODULES = Arrays.asList("math", "numpy", "torch");
+
+  private String importedModuleForIsClose;
+  private Name importedAlias;
 
   @Override
   public void initialize(Context context) {
     context.registerSyntaxNodeConsumer(Tree.Kind.FILE_INPUT,
         ctx -> reachingDefinitionsAnalysis = new ReachingDefinitionsAnalysis(ctx.pythonFile()));
 
+    context.registerSyntaxNodeConsumer(Tree.Kind.IMPORT_NAME,
```

Comment:
Do we need to monitor `IMPORT_FROM` as well? (I don't think so, it's unlikely the `isclose` function has already been imported alone, and worst case we just reimport the whole module next to it - I don't think avoiding that is worth the extra effort)

Reply:
I do not think we do. I feel the rule is quite complex only for the quick fix and handling all the cases can be difficult. I also think it would be ok to just put the import at the top.